src/pkgcore/config/basics.py

```diff
@@ -50,7 +51,13 @@ class ConfigType:
     :ivar allow_unknowns: controls whether unknown settings should error.
     """
 
-    def __init__(self, func_obj):
+    callable: typing.Callable
+    types: dict[str, str]
+    positional: tuple[str]
+    required: tuple[str]
+    allow_unknowns: bool
```

Comment:
Here, opposed to my previous comment, I like the definitions, as `__init__` is very complex so better define it previously.

Reply:
A question: I'm using `typing.Callable` here, and the usage of it- to me- feels like a `typing.Any`.  I'm stating "it's something we invoke", but I give no constraints as to signature of the functor passed.

I assume the typing tools aren't capable of inferring that signature, thus my plan was to go back and add explicitly typed callable's to allow checking to be performed.  I can't say I'm enjoying the python syntax for doing this, but I figure it's best we constrain callback/functor's like this since they've historically been points where testing can't fully flex it.

What's y'alls thoughts here?